src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,183 @@
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        T oldValue = current.value;
+        current.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index is out of bound");
+        }
+        Node<T> current = head;
+        if (index == 0) {
+            final T removeValue = head.value;
+            head = head.next;
+            if (head != null) {
+                head.prev = null;
+            } else {
+                tail = null;
+            }
+            size--;
+            return removeValue;
+        }
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        final T removedValue = current.value;
+        if (current.prev != null) {
+            current.prev.next = current.next;
+        }
+        if (current.next != null) {
+            current.next.prev = current.prev;
+        }
+        if (current == tail) {
+            tail = current.prev;
+        }
+        size--;
+        return removedValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        while (current != null) {
+            if (current.value == null && object == null) {
+                if (current.prev != null) {
+                    current.prev.next = current.next;
+                }
+                if (current.next != null) {
+                    current.next.prev = current.prev;
+                }
+                if (current == tail) {
+                    tail = current.prev;
+                }
+                size--;
+                return true;
+            }
+            if (current.value != null && current.value.equals(object)) {
+                if (current == head) {
+                    remove(0);
+                } else {
+                    if (current.prev != null) {
+                        current.prev.next = current.next;
+                    }
+                    if (current.next != null) {
+                        current.next.prev = current.prev;
+                    }
+                    if (current == tail) {
+                        tail = current.prev;
+                    }
+                    size--;
+                }
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.prev = prev;
+            this.next = next;
+        }
```

Comment:
Don't use getters and setters in class Node. Since `Node<T>` is going to be an inner class, you can directly access the private fields of `Node` from `MyLinkedList`.